server/service/src/report/report_service.rs

```diff
@@ -316,25 +333,101 @@ impl ReportServiceTrait for ReportService {}
 pub const MAX_LIMIT: u32 = 1000;
 pub const MIN_LIMIT: u32 = 1;
 
-fn get_report(ctx: &ServiceContext, id: &str) -> Result<Report, RepositoryError> {
-    ReportRepository::new(&ctx.connection)
-        .query_by_filter(ReportFilter::new().id(EqualFilter::equal_to(id)))?
+#[derive(Debug)]
+pub enum GetReportError {
+    // TODO add more detail about specific report? Breaking out of iterator will prevent listing all reports
+    TranslationError,
+    RepositoryError(RepositoryError),
+}
+
+fn get_report(
+    ctx: &ServiceContext,
+    translation_service: &Box<Localisations>,
+    user_language: String,
+    id: &str,
+) -> Result<Report, GetReportError> {
+    let report = ReportRepository::new(&ctx.connection)
+        .query_by_filter(ReportFilter::new().id(EqualFilter::equal_to(id)))
+        .map_err(|e| GetReportError::RepositoryError(e))?
         .pop()
-        .ok_or(RepositoryError::NotFound)
+        .ok_or(GetReportError::RepositoryError(RepositoryError::NotFound))?;
+
+    let translated_schema = if let Some(argument_schema) = report.argument_schema {
+        Some(
+            translate_schema(argument_schema, translation_service, &user_language)
+                .map_err(|_e| GetReportError::TranslationError)?,
+        )
+    } else {
+        None
+    };
+
+    Ok(Report {
+        report_row: report.report_row,
+        argument_schema: translated_schema,
+    })
+}
+
+#[derive(Debug)]
+pub enum GetReportsError {
+    // TODO add more detail about specific report? Breaking out of iterator will prevent listing all reports
+    TranslationError,
+    ListError(ListError),
 }
 
 fn query_reports(
     ctx: &ServiceContext,
+    translation_service: &Box<Localisations>,
+    user_language: String,
     pagination: Option<PaginationOption>,
     filter: Option<ReportFilter>,
     sort: Option<ReportSort>,
-) -> Result<Vec<Report>, ListError> {
+) -> Result<Vec<Report>, GetReportsError> {
     let repo = ReportRepository::new(&ctx.connection);
-    let pagination = get_default_pagination(pagination, MAX_LIMIT, MIN_LIMIT)?;
+    let pagination = get_default_pagination(pagination, MAX_LIMIT, MIN_LIMIT)
+        .map_err(GetReportsError::ListError)?;
     let filter = filter
         .unwrap_or_default()
         .r#type(ReportType::OmSupply.equal_to());
-    Ok(repo.query(pagination, Some(filter.clone()), sort)?)
+
+    let reports = repo
+        .query(pagination, Some(filter.clone()), sort)
+        .map_err(|e| GetReportsError::ListError(ListError::DatabaseError(e)))?;
+    let reports = reports
+        .into_iter()
+        .map(|r| {
+            let translated_schema = if let Some(argument_schema) = r.argument_schema {
+                Some(
+                    translate_schema(argument_schema, translation_service, &user_language)
+                        .map_err(|_e| GetReportsError::TranslationError)?,
```

Comment:
Failing on translate_schema will break out of the mapping, and return an TranslationError which is not specific to the report that broke.

Because we added a fallback value to the translate function, I am guessing that adding an error specific to the report is not necessary. 

But also thought it would be good to not list errors resulting from here as RepositoryErrors. So I have added a TranslationError enum for this case.